Nodejs/Product/TestAdapter/PackageJsonTestExecutor.cs

```diff
@@ -0,0 +1,187 @@
+            try
+            {
+                var testEvent = JsonConvert.DeserializeObject<TestEvent>(line);
+                // Extract test from list of tests
+                var tests = this.currentTests.Where(n => n.DisplayName == testEvent.title);
+                if (tests.Count() > 0)
+                {
+                    switch (testEvent.type)
+                    {
+                        case "test start":
+                            {
+                                this.currentResult = new TestResult(tests.First())
+                                {
+                                    StartTime = DateTimeOffset.Now
+                                };
+                                this.frameworkHandle.RecordStart(tests.First());
+                            }
+                            break;
+                        case "result":
+                            {
+                                RecordEnd(this.frameworkHandle, tests.First(), this.currentResult, testEvent.result);
+                            }
+                            break;
+                        case "pending":
+                            {
+                                this.currentResult = new TestResult(tests.First());
+                                RecordEnd(this.frameworkHandle, tests.First(), this.currentResult, testEvent.result);
+                            }
+                            break;
+                    }
+                }
+                else if (testEvent.type == "suite end")
+                {
+                    this.currentResultObject = testEvent.result;
+                }
+            }
+            catch (JsonReaderException)
+            {
+                // Often lines emitted while running tests are not test results, and thus will fail to parse above
+            }
+        }
+
+        private void RecordEnd(IFrameworkHandle frameworkHandle, TestCase test, TestResult result, ResultObject resultObject)
+        {
+            var standardOutputLines = resultObject.stdout.Split('\n');
+            var standardErrorLines = resultObject.stderr.Split('\n');
+
+            if (resultObject.pending == true)
+            {
+                result.Outcome = TestOutcome.Skipped;
+            }
+            else
+            {
+                result.EndTime = DateTimeOffset.Now;
+                result.Duration = result.EndTime - result.StartTime;
+                result.Outcome = resultObject.passed ? TestOutcome.Passed : TestOutcome.Failed;
+            }
+
+            result.Messages.Add(new TestResultMessage(TestResultMessage.StandardOutCategory, string.Join(Environment.NewLine, standardOutputLines)));
+            result.Messages.Add(new TestResultMessage(TestResultMessage.StandardErrorCategory, string.Join(Environment.NewLine, standardErrorLines)));
+            result.Messages.Add(new TestResultMessage(TestResultMessage.AdditionalInfoCategory, string.Join(Environment.NewLine, standardErrorLines)));
```

Comment:
>standardErrorLines [](start = 129, length = 18)

We're printing this to two different categories?  If that's correct, consider extracting a local for the joined text.